src/main/java/com/powsybl/openloadflow/ac/AbstractTransformerVoltageControlOuterLoop.java

```diff
@@ -12,41 +12,46 @@
 import com.powsybl.openloadflow.network.LfBranch;
 import com.powsybl.openloadflow.network.LfNetwork;
 import com.powsybl.openloadflow.network.PiModel;
-import com.powsybl.openloadflow.network.TransformerVoltageControl;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
 /**
  * @author Anne Tilloy <anne.tilloy at rte-france.com>
  */
 public abstract class AbstractTransformerVoltageControlOuterLoop implements OuterLoop {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractTransformerVoltageControlOuterLoop.class);
 
-    protected OuterLoopStatus roundVoltageRatios(LfNetwork network) {
+    protected static List<LfBranch> getControllerBranches(LfNetwork network) {
+        return network.getBranches()
+                .stream().filter(branch -> !branch.isDisabled() && branch.isVoltageController())
+                .collect(Collectors.toList());
+    }
+
+    protected OuterLoopStatus roundVoltageRatios(OuterLoopContext context) {
         OuterLoopStatus status = OuterLoopStatus.STABLE;
-        for (LfBranch branch : network.getBranches()) {
-            TransformerVoltageControl voltageControl = branch.getVoltageControl().orElse(null);
-            if (voltageControl != null) {
-                branch.setVoltageControlEnabled(false);
-
-                // round the rho shift to the closest tap
-                PiModel piModel = branch.getPiModel();
-                double r1Value = piModel.getR1();
-                piModel.roundR1ToClosestTap();
-                double roundedR1Value = piModel.getR1();
-                LOGGER.trace("Round voltage ratio of '{}': {} -> {}", branch.getId(), r1Value, roundedR1Value);
-
-                status = OuterLoopStatus.UNSTABLE;
-            }
+        for (LfBranch controllerBranch : getControllerBranches(context.getNetwork())) {
+            controllerBranch.setVoltageControlEnabled(false);
+
+            // round the rho shift to the closest tap
+            PiModel piModel = controllerBranch.getPiModel();
+            double r1Value = piModel.getR1();
+            piModel.roundR1ToClosestTap();
+            double roundedR1Value = piModel.getR1();
+            LOGGER.trace("Round voltage ratio of '{}': {} -> {}", controllerBranch.getId(), r1Value, roundedR1Value);
+
+            status = OuterLoopStatus.UNSTABLE;
         }
         return status;
     }
 
     @Override
     public void cleanup(OuterLoopContext context) {
-        for (LfBranch branch : context.getNetwork().getBranches()) {
```

Comment:
Why do you need this clean up? It is already done through the rounded of ratios, no?